### DISCUSSION TITLE
File upload refactor to support multi-file S3 uploads

### DISCUSSION THREAD

**Discussion by jameslkingsley** [livewire/livewire#2807] [3. Feature ideas]
I recently had to implement a multi-file uploader to S3 at work. Since Livewire doesn't support multi-file uploads to S3 out of the box, I wrote a custom directive that handles it near enough the same way.

I want to PR this to Livewire, and with it a refactored file upload system that's a bit more configurable. This is more of a question to @calebporzio — is this something you want or is there already something in the works/major changes that would affect this?

**Comment by calebporzio** [comment]
Thanks James! There is nothing in the works right now. I'm open to looking at a PR for it. No promise to merge but would definitely review a WIP PR for it.

**Reply by jameslkingsley** [reply]
Sent! https://github.com/livewire/livewire/pull/2816
